### ISSUE TITLE
[Accessibility]: Screen readers do not read currently focused line when using Alt+F5/Alt+Shift+F5

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#164988]
CC @isidorn

## System Info

Version: 1.73.0-insider (system setup)
Commit: 3544aabd5a2b0da2cd3a6f3fa3f00759aa9d9709
Date: 2022-10-27T21:34:51.585Z
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes

## Reproducible Steps

1. Load screen reader (e.g., NVDA).

2. Open any diffs between two files (or PR).

3. Press `Alt+F5` to go to the next diff and `Alt+Shift+F5` to go to the previous diff.

4. See if screen reader reads aloud the currently focused line: it does not say anything.

## Expected Behaviors

Focused line needs to be auto-announced.

**Comment by isidorn** [comment]
@jooyoungseo thanks for filling this issue.
Assigning first to @meganrogge for an initial investigation.

**Comment by meganrogge** [comment]
can reproduce

**Comment by meganrogge** [comment]
was this working historically? @isidorn @jooyoungseo perhaps we're dealing with a regression?

The behavior currently is the line near the change gets read, but sometimes it doesn't even include the change

https://user-images.githubusercontent.com/29464607/204812067-f343fa22-3736-4981-9dc6-ed4c2a7d53a1.mov

**Comment by jooyoungseo** [comment]
@meganrogge  -- No, this is not an inadvertent regression b/c it has never worked as far as I experienced.

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by rperez030** [comment]
@isidorn  @meganrogge  I wonder what the right way should be to address a request like this in the context of VS Code if it moves forward. what I mean is that, by design, screen readers, or at least NVDA and JAWS, do not read changes in cursor position automatically unless the change is initiated by the user using the arrow keys, so when we press alt +F5, the cursor moves to the line in conflict. This is detected by the screen reader and the new line appears in the Braille display if there is one connected, but the screen reader does not read the line automatically. This can be easily tested with another text editor such as Notepad using the Find next feature which moves the cursor but doesn't cause the screen reader to auto announce. One possibility would be to auto select the line in conflict which would cause the screen reader to read. This is what Microsoft word does, for example.

**Comment by meganrogge** [comment]
@rperez030 thanks for that insight. I plan to explore this in January, so will consider that approach

**Comment by isidorn** [comment]
I like the auto selection idea.

**Comment by meganrogge** [comment]
I just merged the feature that plays audio cues when navigating between changes yesterday https://github.com/microsoft/vscode/issues/166518.

In working on this request today, I notice that the audio cues don't get played anymore when the line content is read (in oss based on my branch where I'm working on this). Is that expected/ what you would want @jooyoungseo and @rperez030 or would you prefer to still have them play - either before or after the content is read?

I'm not sure if it is possible to support both, but want to know what you think before further investigating.

**Comment by jooyoungseo** [comment]
Signaling audio cues before content is read would be preferred. Otherwise, we need to wait until the content is being read to get any audio hint. I am not sure whether it's technically possible to trigger the audio cues right before any screen reader reading event.

**Comment by meganrogge** [comment]
@jooyoungseo When I add a 500 ms delay between the call to play the audio cue and the update of selection/content to be read, it works. This is because the audio cue can take up to 1 second to load. 

Is that delay worth it in your opinion?

**Comment by jooyoungseo** [comment]
@meganrogge -- That makes sense to me. Did you add that delay to all the audio cues? I can share my experience once it's available in insider version tomorrow.

**Comment by meganrogge** [comment]
I just added it to the ones for which the line will be read (go to next/previous difference). When my PR is merged, this issue will be closed. Once the insider's released label is attached, it would be great if you could test it and let me know what you think. I think it's a good experience to have both as you suggested
